hypertrace-view-generator/hypertrace-view-generator/src/main/java/org/hypertrace/viewgenerator/generators/SpanEventViewGenerator.java

```diff
@@ -124,12 +134,55 @@ private Event getApiEntrySpanForExitSpan(
         .orElse(null);
   }
 
+  /**
+   * An api_node represents an api call in the trace
+   * It consists of api_entry span and multiple api_exit and internal spans
+   * <p>
+   * This method computes the count of exit calls for any given api (identified by api_entry_span)
+   * This count is a composition of 2 things
+   * 1. link between api_exit_span in api_node to api_entry_span (child of api_exit_span) in another api_node
+   * 2. exit calls to backend from api_exit_span in api_node
+   */
+  Map<ByteBuffer, Integer> computeApiExitCallCount(StructuredTrace trace) {
+    // api_node_index -> api exit call count
+    Map<Integer, AtomicInteger> apiNodeExitCallCount = Maps.newHashMap();
+    ApiTraceGraph apiTraceGraph = ViewGeneratorState.getApiTraceGraph(trace);
+
+    // this set contains events which are the source of any api_node_edge
+    Set<ByteBuffer> callerExitEvents = Sets.newHashSet();
+    for (ApiNodeEventEdge apiNodeEventEdge : apiTraceGraph.getApiNodeEventEdgeList()) {
+      // there will be at max 1 call between any two api_node
+      apiNodeExitCallCount.computeIfAbsent(
+          apiNodeEventEdge.getSrcApiNodeIndex(), v -> new AtomicInteger(0)).incrementAndGet();
+      // srcEventIndex won't be null
+      callerExitEvents.add(trace.getEventList().get(apiNodeEventEdge.getSrcEventIndex()).getEventId());
+    }
+
+    // event -> api exit call count for the corresponding api_node
+    Map<ByteBuffer, Integer> eventToApiExitCallCount = Maps.newHashMap();
+
+    for (int index = 0; index < apiTraceGraph.getNodeList().size(); index++) {
+      ApiNode<Event> apiNode = apiTraceGraph.getNodeList().get(index);
+      // {@link ApiTraceGraph#getApiNodeEventEdgeList()} doesn't consists of edges to backend, so compute them explicitly
+      int backendExitCallCountForApiNode = (int) apiNode
+          .getExitApiBoundaryEvents().stream()
+          .filter(exitEvent -> !callerExitEvents.contains(exitEvent.getEventId())).count();
+
+      int totalExitCallCount =
+          apiNodeExitCallCount.getOrDefault(index, new AtomicInteger(0)).get()
+              + backendExitCallCountForApiNode;
+      apiNode.getEvents().forEach(e -> eventToApiExitCallCount.put(e.getEventId(), totalExitCallCount));
+    }
+    return eventToApiExitCallCount;
```

Comment:
Still feels to me like this could go into the api trace graph. Conceptually, we're just asking for - given an event, what api call does it belong to, then given an api call, what are its connections to backends and other api calls, right.

So I guess I'd think the api would make it look something like
```java
graph.getApiNode(eventId).getApiNodeEdges().size() + graph.getApiNode(eventId).getBackendEdges().size();
```

I don't have enough context to know what kind of change this would be, but it does seem like we're just trying pull data out of the graph here that ultimately should live in it.

Reply:
I agree, I have mentioned in https://github.com/hypertrace/hypertrace-ingester/issues/139, about the need to move some of these logic to ApiTraceGraph. 
Currently for some reason we are not calculating backend edge in ApiTraceGraph, first I need to check on that and take up appropriate refactoring. 
However, I would not prefer taking up such refactoring work as part of this particular change. What needs to be moved into ApiTraceGraph needs to be carefully thought and if needed some existing structures should also be refactored.

Reply:
ok